envoy/datadog_checks/envoy/metrics.py

```diff
@@ -140,6 +140,27 @@
         ),
         'method': 'gauge',
     },
+    'control_plane.connected_state': {
+        'tags': (
+            (),
+            (),
+        ),
+        'method': 'gauge',
+    },
+    'control_plane.pending_requests': {
+        'tags': (
+            (),
+            (),
+        ),
+        'method': 'gauge',
+    },
+    'control_plane.rate_limit_enforced': {
+        'tags': (
+            (),
+            (),
+        ),
+        'method': 'monotonic_count',
+    },
```

Comment:
Thanks for this PR. Looks good to me overall.

Could you manage to test all/some of those new metrics in e2e ? By adding metrics here https://github.com/DataDog/integrations-core/blob/81792d0e48f8083fb288a411662ba4f1a39ba894/envoy/tests/test_e2e.py#L12-L161

That would probably need some changes in setup files here: https://github.com/DataDog/integrations-core/tree/81792d0e48f8083fb288a411662ba4f1a39ba894/envoy/tests/docker/default

Reply:
I've pushed another commit which adds this, but I'm seeing test failures with the newly-added metrics despite verifying that those metrics are properly parsed in the unit tests from my last commit, and verifying that the Envoy instance is now reporting those metrics with the added controlplane implementation. Do you know what might be causing that?